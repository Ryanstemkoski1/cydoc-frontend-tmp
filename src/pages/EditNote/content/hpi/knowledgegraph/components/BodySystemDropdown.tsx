import 'pages/EditNote/content/hpi/knowledgegraph/css/Button.css';
import React from 'react';
import { Button } from 'semantic-ui-react';
import '../HPI.css';
import '../css/Button.css';
import ChiefComplaintsButton from './ChiefComplaintsButton';

interface BodySystemDropdownProps {
    diseasesList: string[];
    name: string;
}

interface BodySystemDropdownState {
    selected: boolean;
}

const imgToRender: { [key: string]: any } = {
    Neurology: '/images/brain.svg',
    Psychiatry: '/images/psychiatry.svg',
    'Ophtho/ENT': '/images/heent.svg',
    Pulmonology: '/images/respiratory.svg',
    Cardiology: '/images/heart.svg',
    'General Medicine': '/images/genmed.svg',
    Gastroenterology: '/images/intestines.svg',
    Rheumatology: '/images/immunology.svg',
    Dermatology: '/images/dermatologic.svg',
    'Ob/Gyn': '/images/kidneys.svg',
    Endocrinology: '/images/endocrine.svg',
    Pediatrics: '/images/pediatrics.svg',
    Orthopedics: '/images/musculoskeletal.svg',
    Favorites: '/images/star.svg',
    'Heme/Onc': '/images/blooddrop.svg',
};

class BodySystemDropdown extends React.Component<
    BodySystemDropdownProps,
    BodySystemDropdownState
> {
    constructor(props: BodySystemDropdownProps) {
        super(props);
        this.state = {
            selected: false, //Set to false so that dropdown is closed by default
        };
    }

    render(): React.ReactNode {
        const { name, diseasesList } = this.props;

        //Defining outlier conditions
        //for styling body system dropdown buttons including ...
        //if the screen size is mobile -> change button size and layout
        //if the BodySystem is Cardiovascular/Hematologic -> abbreviate to Cards/Heme
        //if the BodySystem is Neurologic/Psychiatric -> abbreviate to Neuro/Psych

        //Setting default render style to computer but changing to mobile if needed
        const styleToRender = 'hpi-disease-button';

        return (
            <div>
                <Button
                    className={`hpi-chiefcomplaint-button ${styleToRender}`}
                    onClick={() =>
                        this.setState({ selected: !this.state.selected })
                    }
                    active={this.state.selected}
                >
                    {/* <Icon name='dropdown' /> */}

                    <div className='hpi-disease-button-content-outer'>
                        <div className='hpi-disease-button-content-icon'>
                            <img
                                className={'hpi-icons'}
                                src={imgToRender[name]}
                            />
                        </div>
                        <div className='hpi-disease-button-content-name'>
                            {name}
                        </div>
                    </div>
                </Button>
                <div className='diseases-array btn-wrap'>
                    {this.state.selected
                        ? diseasesList
                              .filter((disease) => disease !== 'HIDDEN')
                              .map((disease) => (
                                  <ChiefComplaintsButton
                                      key={disease}
                                      name={disease}
                                  />
                              ))
                        : []}
                </div>
            </div>
        );
    }
}

export default BodySystemDropdown;
