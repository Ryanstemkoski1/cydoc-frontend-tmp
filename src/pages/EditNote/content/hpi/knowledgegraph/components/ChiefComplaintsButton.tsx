import axios from 'axios';
import ToggleButton from 'components/tools/ToggleButton/ToggleButton';
import { graphClientURL } from 'constants/api.js';
import { GraphData } from 'constants/hpiEnums';
import { getSelectedChiefCompliants } from 'hooks/useSelectedChiefComplaints';
import { initialSurveyProps } from 'pages/HPI/ChiefComplaintSelection/CCSelection';
import React from 'react';
import { connect } from 'react-redux';
import {
    selectChiefComplaint,
    SelectChiefComplaintAction,
} from 'redux/actions/chiefComplaintsActions';
import {
    processKnowledgeGraph,
    ProcessKnowledgeGraphAction,
} from 'redux/actions/hpiActions';
import { CurrentNoteState } from 'redux/reducers';
import { selectActiveItem } from 'redux/selectors/activeItemSelectors';
import { selectChiefComplaintsState } from 'redux/selectors/chiefComplaintsSelectors';
import {
    selectInitialPatientSurvey,
    selectPatientViewState,
} from 'redux/selectors/userViewSelectors';
import { ChiefComplaintsProps, HpiHeadersProps } from '../HPIContent';
import './BodySystemDropdown';
interface ChiefComplaintsButtonProps {
    name: string;
}

class ChiefComplaintsButton extends React.Component<Props> {
    getData = async (chiefComplaint: string) => {
        if (!chiefComplaint) {
            return;
        }
        const response = await axios.get(
            graphClientURL + '/graph/category/' + chiefComplaint + '/4'
        );
        this.props.processKnowledgeGraph(response.data);
    };
    render() {
        const {
            selectChiefComplaint,
            name,
            chiefComplaints,
            hpiHeaders,
            patientView,
            activeItem,
            userSurveyState,
        } = this.props;
        return (
            <ToggleButton
                className='tag_text btn-space'
                active={name in chiefComplaints}
                condition={name}
                title={
                    patientView && name in hpiHeaders.parentNodes
                        ? hpiHeaders.parentNodes[name].patientView
                        : name
                }
                onToggleButtonClick={(e: any) => {
                    const pinnedChiefComplaints = Object.keys(
                        userSurveyState.nodes['6'].response ?? {}
                    );
                    const pinnedSelectedChiefComplaints =
                        getSelectedChiefCompliants(chiefComplaints).filter(
                            (item) => pinnedChiefComplaints.includes(item)
                        );

                    if (
                        activeItem === 'CCSelection' &&
                        !pinnedSelectedChiefComplaints.includes(name) &&
                        pinnedSelectedChiefComplaints.length >= 3
                    ) {
                        return;
                    }

                    selectChiefComplaint(name);

                    if (!(name in hpiHeaders?.parentNodes)) {
                        console.error(
                            `Chief Complaint named '${name}' is not present in the Knowledge Graph API response, SYSTEM MIGHT FAIL DUE TO THIS`
                        );
                    }

                    this.getData(
                        Object.keys(hpiHeaders?.parentNodes?.[name])?.[0]
                    );
                }}
            />
        );
    }
}

interface DispatchProps {
    selectChiefComplaint: (disease: string) => SelectChiefComplaintAction;
    processKnowledgeGraph: (
        graphData: GraphData
    ) => ProcessKnowledgeGraphAction;
}

export interface PatientViewProps {
    patientView: boolean;
}

interface ActiveItemProps {
    activeItem: string;
}

const mapStateToProps = (
    state: CurrentNoteState
): ChiefComplaintsProps &
    initialSurveyProps &
    HpiHeadersProps &
    PatientViewProps &
    ActiveItemProps => ({
    chiefComplaints: selectChiefComplaintsState(state),
    hpiHeaders: state.hpiHeaders,
    patientView: selectPatientViewState(state),
    activeItem: selectActiveItem(state),
    userSurveyState: selectInitialPatientSurvey(state),
});

type Props = DispatchProps &
    initialSurveyProps &
    ChiefComplaintsButtonProps &
    ChiefComplaintsProps &
    HpiHeadersProps &
    PatientViewProps &
    ActiveItemProps;

const mapDispatchToProps = {
    selectChiefComplaint,
    processKnowledgeGraph,
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(ChiefComplaintsButton);
