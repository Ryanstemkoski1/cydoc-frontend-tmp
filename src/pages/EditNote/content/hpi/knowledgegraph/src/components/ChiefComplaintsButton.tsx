import React from 'react';
import './BodySystemDropdown';
import { DoctorView } from 'constants/hpiEnums';
import {
    selectChiefComplaint,
    SelectChiefComplaintAction,
} from 'redux/actions/chiefComplaintsActions';
import { ChiefComplaintsState } from 'redux/reducers/chiefComplaintsReducer';
import { CurrentNoteState } from 'redux/reducers';
import { connect } from 'react-redux';

interface ChiefComplaintsButtonProps {
    name: DoctorView;
}

class ChiefComplaintsButton extends React.Component<Props> {
    render() {
        const { chiefComplaints, selectChiefComplaint, name } = this.props;
        const selected = chiefComplaints.includes(name);
        return (
            <button
                className='tag_text'
                style={{
                    backgroundColor: selected ? 'lightslategrey' : 'whitesmoke',
                    color: selected ? 'white' : 'black',
                }}
                onClick={(): SelectChiefComplaintAction =>
                    selectChiefComplaint(name)
                }
            >
                {name}
            </button>
        );
    }
}

export interface ChiefComplaintsProps {
    chiefComplaints: ChiefComplaintsState;
}

interface DispatchProps {
    selectChiefComplaint: (disease: DoctorView) => SelectChiefComplaintAction;
}

const mapStateToProps = (state: CurrentNoteState): ChiefComplaintsProps => ({
    chiefComplaints: state.chiefComplaints,
});

type Props = ChiefComplaintsProps & DispatchProps & ChiefComplaintsButtonProps;

const mapDispatchToProps = {
    selectChiefComplaint,
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(ChiefComplaintsButton);
