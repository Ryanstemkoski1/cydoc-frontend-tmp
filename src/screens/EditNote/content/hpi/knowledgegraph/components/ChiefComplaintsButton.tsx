import axios from 'axios';
import ToggleButton from '@components/tools/ToggleButton/ToggleButton';
import { graphClientURL } from '@constants/api';
import { getSelectedChiefComplaints } from '@hooks/useSelectedChiefComplaints';
import React from 'react';
import { ConnectedProps, connect } from 'react-redux';
import { setChiefComplaint } from '@redux/actions/chiefComplaintsActions';
import { processKnowledgeGraph } from '@redux/actions/hpiActions';
import { CurrentNoteState } from '@redux/reducers';
import { selectActiveItem } from '@redux/selectors/activeItemSelectors';
import { selectChiefComplaintsState } from '@redux/selectors/chiefComplaintsSelectors';
import {
    selectInitialPatientSurvey,
    selectPatientViewState,
} from '@redux/selectors/userViewSelectors';
import './BodySystemDropdown';

interface OwnProps {
    name: string;
}

type ReduxProps = ConnectedProps<typeof connector>;

type Props = ReduxProps & OwnProps;

class ChiefComplaintsButton extends React.Component<Props> {
    getData = async (chiefComplaint: string) => {
        if (!chiefComplaint) {
            return;
        }
        const response = await axios.get(
            graphClientURL + '/graph/category/' + chiefComplaint + '/4'
        );
        this.props.processKnowledgeGraph(response.data);
    };
    render() {
        const {
            setChiefComplaint,
            name,
            chiefComplaints,
            hpiHeaders,
            patientView,
            activeItem,
            userSurveyState,
        } = this.props;
        return (
            <ToggleButton
                className='tag_text btn-space'
                active={name in chiefComplaints}
                condition={name}
                title={
                    patientView && name in hpiHeaders.parentNodes
                        ? hpiHeaders.parentNodes[name].patientView
                        : name
                }
                onToggleButtonClick={() => {
                    const pinnedChiefComplaints = Object.keys(
                        userSurveyState.nodes['6'].response ?? {}
                    );
                    const pinnedSelectedChiefComplaints =
                        getSelectedChiefComplaints(chiefComplaints).filter(
                            (item) => pinnedChiefComplaints.includes(item)
                        );

                    if (
                        activeItem === 'CCSelection' &&
                        !pinnedSelectedChiefComplaints.includes(name) &&
                        pinnedSelectedChiefComplaints.length >= 3
                    ) {
                        return;
                    }

                    setChiefComplaint(name);

                    if (!(name in hpiHeaders?.parentNodes)) {
                        console.error(
                            `Chief Complaint named '${name}' is not present in the Knowledge Graph API response, SYSTEM MIGHT FAIL DUE TO THIS`
                        );
                    }

                    this.getData(
                        Object.keys(hpiHeaders?.parentNodes?.[name])?.[0]
                    );
                }}
            />
        );
    }
}

const mapStateToProps = (state: CurrentNoteState) => ({
    chiefComplaints: selectChiefComplaintsState(state),
    hpiHeaders: state.hpiHeaders,
    patientView: selectPatientViewState(state),
    activeItem: selectActiveItem(state),
    userSurveyState: selectInitialPatientSurvey(state),
});

const mapDispatchToProps = {
    setChiefComplaint,
    processKnowledgeGraph,
};

const connector = connect(mapStateToProps, mapDispatchToProps);

export default connector(ChiefComplaintsButton);
