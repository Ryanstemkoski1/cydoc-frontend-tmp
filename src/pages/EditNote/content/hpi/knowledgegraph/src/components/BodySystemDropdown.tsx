import React from 'react';
import { Button } from 'semantic-ui-react';
import '../css/Button.css';
import ChiefComplaintsButton from './ChiefComplaintsButton';
import '../../HPI.css';
import { BodySystemNames, DoctorView } from 'constants/hpiEnums';
import { NOTE_PAGE_MOBILE_BP } from 'constants/breakpoints';
import brain from '../icons/brain.png';
import pain from '../icons/pain.png';
import heent from '../icons/heent.png';
import respiratory from '../icons/respiratory.png';
import heart from '../icons/heart.png';
import yoga from '../icons/yoga.png';
import intestines from '../icons/intestines.png';
import immunology from '../icons/immunology.png';
import dermatologic from '../icons/dermatologic.png';
import kidneys from '../icons/kidneys.png';
import genitourinary from '../icons/genitourinary.png';

interface BodySystemDropdownProps {
    diseasesList: DoctorView[];
    name: BodySystemNames;
}

interface BodySystemDropdownState {
    selected: boolean;
}

const imgToRender: { [key: string]: any } = {
    'Neurologic/Psychiatric': brain,
    Pain: pain,
    HEENT: heent,
    Respiratory: respiratory,
    'Cardiovascular/Hematologic': heart,
    'General/Lifestyle': yoga,
    Gastrointestinal: intestines,
    Immune: immunology,
    Dermatologic: dermatologic,
    Genitourinary: kidneys,
    Endocrine: genitourinary,
};

class BodySystemDropdown extends React.Component<
    BodySystemDropdownProps,
    BodySystemDropdownState
> {
    constructor(props: BodySystemDropdownProps) {
        super(props);
        this.state = {
            selected: true,
        };
    }

    render(): React.ReactNode {
        const { name, diseasesList } = this.props;
        const isMobile = window.innerWidth < NOTE_PAGE_MOBILE_BP;
        const mobileCardio = isMobile && imgToRender[name] == heart;
        const normalCardio = !isMobile && imgToRender[name] == heart;
        let styleToRender = 'hpi-disease-button';

        if (isMobile) {
            styleToRender = 'mobile-hpi-disease-button';
        }
        if (mobileCardio) {
            styleToRender = 'cardiovascular-mobile';
        }
        if (normalCardio) {
            styleToRender = 'cardiovascular-normal';
        }

        return (
            <div>
                <Button
                    basic
                    className={styleToRender}
                    onClick={() =>
                        this.setState({ selected: !this.state.selected })
                    }
                >
                    {/* <Icon name='dropdown' /> */}
                    <img
                        className={isMobile ? 'mobile-hpi-icons' : 'hpi-icons'}
                        src={imgToRender[name]}
                    />
                    {name}
                </Button>
                <div className='diseases-array'>
                    {this.state.selected
                        ? diseasesList
                              .filter((disease) => disease !== 'HIDDEN')
                              .map((disease) => (
                                  <ChiefComplaintsButton
                                      key={disease}
                                      name={disease}
                                  />
                              ))
                        : []}
                </div>
            </div>
        );
    }
}

export default BodySystemDropdown;
