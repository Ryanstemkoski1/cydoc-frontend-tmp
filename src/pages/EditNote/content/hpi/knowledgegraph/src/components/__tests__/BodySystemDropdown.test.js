import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import BodySystemDropdown from '../BodySystemDropdown';
import ChiefComplaintsButton from '../ChiefComplaintsButton';

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe('ChiefComplaintsButton', () => {
    const wrapper = shallow(
        <BodySystemDropdown
            name='Cardiovascular'
            diseasesList={[
                'Abdominal Pain',
                'Aging Challenges and Dementia',
                'Alcohol Use',
            ]}
        />
    );
    it('placeholder so suite can run', () => {});
    // // TODO: Fix below tests
    // it('renders body system dropdown', () => {
    //     expect(wrapper).toMatchSnapshot();
    //     expect(wrapper.state().selected).toBeTruthy();
    // });
    // it('drops dropdown when clicked and unclicked', () => {
    //     wrapper.find('.hpi-disease-button').simulate('click');
    //     expect(wrapper.state().selected).toBeFalsy();
    //     wrapper.find('.hpi-disease-button').simulate('click');
    //     expect(wrapper.state().selected).toBeTruthy();
    // });
    // it('opens disease array when clicked and unclicked', () => {
    //     expect(wrapper.find(ChiefComplaintsButton).length).toEqual(3);
    //     wrapper.find('.hpi-disease-button').simulate('click');
    //     expect(
    //         wrapper.containsMatchingElement(<ChiefComplaintsButton />)
    //     ).toBeFalsy();
    // });
});
