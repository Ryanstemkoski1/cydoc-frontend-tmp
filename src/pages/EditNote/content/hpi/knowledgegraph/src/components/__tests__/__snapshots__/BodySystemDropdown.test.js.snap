// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChiefComplaintsButton renders body system dropdown 1`] = `
<div>
  <Button
    active={false}
    as="button"
    className="hpi-chiefcomplaint-button hpi-disease-button"
    onClick={[Function]}
  >
    <div
      className="hpi-disease-button-content-outer"
    >
      <div
        className="hpi-disease-button-content-icon"
      >
        <img
          className="hpi-icons"
        />
      </div>
      <div
        className="hpi-disease-button-content-name"
      >
        Cardiovascular
      </div>
    </div>
  </Button>
  <div
    className="diseases-array"
  />
</div>
`;
